src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -264,18 +265,31 @@ public void SetAll(bool value)
         ** Exceptions: ArgumentException if value == null or
         **             value.Length != this.Length.
         =========================================================================*/
-        public BitArray And(BitArray value)
+        public unsafe BitArray And(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
             if (Length != value.Length)
                 throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);
 
-            for (int i = 0; i < m_array.Length; i++)
+            int i = 0;
+            if (Sse2.IsSupported && m_array.Length >= 4)
             {
-                m_array[i] &= value.m_array[i];
+                fixed (int* leftPtr = m_array)
+                fixed (int* rightPtr = value.m_array)
+                {
+                    for (; i < m_array.Length - 4; i += 4)
+                    {
+                        var leftVec = Sse2.LoadVector128(leftPtr + i);
+                        var rightVec = Sse2.LoadVector128(rightPtr + i);
+                        Sse2.Store(leftPtr + i, Sse2.And(leftVec, rightVec));
+                    }
+                }
             }
 
+            for (; i < m_array.Length; i++)
```

Comment:
We could probably unroll this as well, since we should only have (at most) 3 remaining.

Reply:
@tannergooding I wish JIT could unroll it for me 😄 (https://github.com/dotnet/coreclr/pull/19594?). I'll try and update the benchmarks.

Reply:
It will be not unrolled until limit of IV is constant. you have to unroll it manually if you want.
that partial unrolling support does not unroll dynamic IV which has lots of cost to analyse.

And it will not really affect on performance even after its unrolled.

Reply:
And reason why its just a normal ALU, which has only one port.
If you want to unroll it. You have to make it to use SIMD array.(use difference 2 vector registers) and make it unrolled twice will so make significal changes on performance.

Reply:
Yes, It really does matter its unrolled or not. so its good code :>